draft-dekater-scion-dataplane.md

```diff
@@ -1806,14 +1805,82 @@ However, the path choice of the endpoint may possibly be exploited by an attacke
 
-The SCION IP Gateway (SIG) enables IP packets to be tunnelled over SCION to support the use of applications that are not SCION enabled.
+The SCION IP Gateway (SIG) enables IP packets to be tunnelled over SCION to support communication between hosts that do not run a SCION implementation. A SIG acts as a router from the perspective of IP, whilst acting as SCION endpoint from the perspective of the SCION network. It is typically deployed inside the same AS-internal network as its non-SCION hosts, or at the edge an enterprise network.
 
-An ingress SIG encapsulates IP packets within SCION packets and sends them across a SCION network to an egress SIG. The egress SIG decapsulates the IP packets from the SCION packets and forwards them towards their destination IP address. The SIGs at either end of a tunnel act as routers from the perspective of IP, whilst acting as SCION endpoints from the perspective of the SCION network.
+Tunneling IP traffic over SCION requires a pair of SIGs and it involves the following steps:
 
-Each SIG establishes a session to one or multiple remote SIGs. Each pair of SIGs uses a common tunneling protocol. Each SIG may choose to send SCION packets to a remote SIG in accordance with static IP routes, or by dynamically announcing IP prefixes to each other via a routing protocol. Each SIG may also choose how to send SCION packets based on locally configured policies when multiple remote SIGs are available.
-In addition, the source SIG is responsible for SCION path selection.
+1. A sender sends an IP packet towards an IP destination.
+
+2. The IP packet reaches a SIG in the sender’s network via standard IP routing.
+
+3. Based on the destination IP address, the source SIG determines the destination SIG's ISD-AS, endpoint address. To achieve this, SIGs may be pre-configured with static IP prefix to remote SIG SCION address mappings or they may speak a dynamic routing protocol with the corresponding remote SIG.
+
+4. The SIG encapsulates the original IP packet within a SCION packet and sends it to the remote SIG. If necessary, the source SIG performs a SCION path lookups and selects a SCION path to the destination SIG.
+
+5. The remote SIG receives the SCION packet and decapsulates the original IP packet. It then forwards the packet to the final IP destination using standard local IP routing.
+
+## SIG Framing
+
+IP packets are encapsulated over SCION/UDP into SIG frames. There may be multiple IP packets in a single SIG frame, and a single IP packet may be split into multiple SIG frames.
+
+A pair of SIGs establishes a SIG tunneling session. TODO: how?.  Within each session there may be multiple streams which is useful to distinguish between traffic sent by different SIG instances. For example, if an ingress SIG is restarted, it will create a new Stream ID for each session so the egress SIG will know that the new frame with a new Stream ID does not carry trailing part of the unfinished IP packet from a different stream.
```

Comment:
I admit that I do not know exactly when/why new sessions are established. It is probably an important and not part of the wire format part of the protocol (so-called choreography) for inter-SIG compatibility. Should we attempt at describing it. I can research it in the code.